### ISSUE TITLE
Sticky headers should have a shadow

### ISSUE THREAD

**Issue by vdesmet93** [timehop/sticky-headers-recyclerview#85]
The list headers should get a shadow as soon as they become sticky, because the headers should be above the list, not on the same Z.
Currently, the headers don't have any elevation which looks weird when the Toolbar does have a shadow.

**Comment by djdmbrwsk** [comment]
I implement [header shadows here](https://github.com/djdmbrwsk/sticky-headers-recyclerview/commit/79aea81e38015b6cfbf383930dae13b41e656c8c?diff=unified). However, it needs to be cleaned up a bit.

@jacobtabak do you have any ideas on how to allow users to optionally specify a method to draw the shadow? Right now I just hacked a `GradientDrawable` directly into `onDrawOver()`.

**Comment by jacobtabak** [comment]
I don't think it should be the responsibility of this library to create shadows. However it seems like a good idea to have each header have two states: normal and sticky. That way the developer could decide to add a shadow if necessary for the sticky state.

**Comment by jacobtabak** [comment]
maybe add methods `onCreateStickyHeaderViewHolder` and `onBindStickyHeaderViewHolder` to `StickyRecyclerHeadersDecoration`. 

Not sure... I don't really want to break the API

**Comment by djdmbrwsk** [comment]
Bleh, yeah, extending definitely wouldn't work. And yeah, it would be nice if we didn't have to make a breaking change. I'll give it some more thought.

**Comment by showang** [comment]
I need this feature too...

**Comment by djdmbrwsk** [comment]
@showang checkout my fork https://github.com/djdmbrwsk/sticky-headers-recyclerview/tree/header-shadows
